pappl-retrofit/web-interface.c

```diff
@@ -1066,6 +1066,34 @@ _prSystemWebAddPPD(
 		    // for a PostScript printer we do not worry about
 		    // options with unsufficient PostScript or PJL
 		    // code.
+		    cups_array_t *report;       // Report variable for ppdTest
+		    cups_array_t *file_array;   // List of PPD Files
+		    int result;                 // PPD passed?
+		    char output_string[1024];   // Concatenated string consisting of entire report
+		    int x = 0;                  // Looping var
+		    int y = 0;                  // Looping var
+
+		    cupsArrayAdd(file_array, destpath);
+		    result = ppdTest(0,0,NULL,0,0,0,0,0,1,file_array,&report,NULL,NULL);
+
+                    if (report)
+                    {
+                       for (line = (char *)cupsArrayFirst(report); line; line = (char *)cupsArrayNext(report))
+                       {
+		          x = 0;
+			  while (line[x] != "\0")
+			  {
+			     output_string[y] = line[x];
+			     x++;
+			     y++;
+			  }
+			  output_string[y] = "\n";
+			  y++;
+                       }
+		       papplClientHTMLPrintf(client, "          <div class=\"banner\">%s</div>\n", output_string);
+                    }
+
```

Comment:
Also do not forget `cupsArrayDelete(file_array);`.

Reply:
Yes, I will add it.